### PR TITLE
Read switch_id of fabric switch from config_db

### PR DESCRIPTION
**Pull request by ysmanman**
**What I did**
Read switch_id of fabric switch from config_db

**Why I did it**
Fix https://github.com/sonic-net/sonic-buildimage/issues/18558

**How I verified it**

**Details if related**

### CODE REVIEW

**Review by judyjoseph — COMMENTED** [review]
LGTM, @saksarav-nokia f.y.i

File: orchagent/main.cpp [hunk 1]
```diff
@@ -579,6 +579,28 @@ int main(int argc, char **argv)
...
+
+            if (gVoqMySwitchId < 0)
+            {
+                SWSS_LOG_ERROR("Invalid fabric switch id %d configured", gVoqMySwitchId);
+                exit(EXIT_FAILURE);
+            }
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Fabric switch id is not configured");
```
Comment (prsunny) [inline orchagent/main.cpp:600]: Seems like its a breaking change. You are exiting orchagent if switch_id is not present in db. Is this intenational. @judyjoseph , @arlakshm

Comment (arlakshm) [reply orchagent/main.cpp:600]: @prsunny We have the same condition for linecard asics. In latest SAI, `switch_id` was made mandatory on switch create of fabric asic https://github.com/opencomputeproject/SAI/pull/1793. So, we need this change to pass the correct switch-id. The configuration is expected to have switch_id for both linecard asic and fabric asic.

Comment (prsunny) [reply orchagent/main.cpp:600]: ok, please approve the PR

**Review by arlakshm — DISMISSED** [review]

File: orchagent/main.cpp [hunk 2]
```diff
@@ -579,6 +579,28 @@ int main(int argc, char **argv)
         attr.value.u32 = SAI_SWITCH_TYPE_FABRIC;
         attrs.push_back(attr);
 
+        //Read switch_id from config_db.
+        Table cfgDeviceMetaDataTable(&config_db, CFG_DEVICE_METADATA_TABLE_NAME);
```
Comment (prsunny) [inline orchagent/main.cpp:583]: Can you check if VS tests are adopted for this change for Voq?

**Comment by prsunny** [comment]
@arlakshm , looks like its a genuine VS failure.. please check my comment

**Comment by ysmanman** [comment]
> @arlakshm , looks like its a genuine VS failure.. please check my comment

Thanks @prsunny . I will add switch_id in config_db of virtual chassis.

**Comment by prsunny** [comment]
Can you please fix coverage?

**Comment by ysmanman** [comment]
> Can you please fix coverage?

Hi @prsunny updated PR with unit test.

**Comment by ysmanman** [comment]
@arlakshm @prsunny All checks passed. Please help reviewing and merging the PR.

**Review by arlakshm — APPROVED** [review]

**Pull request merged**